### PR TITLE
serverless-cdk-plugin: Add ServerlessConstruct typing

### PR DESCRIPTION
**Pull request by GuillaumeLagrange**
This PR updates root tsconfig's node resolution to node16 by default.
We do this to support the exports feature, as per this issue: 
https://github.com/microsoft/TypeScript/issues/33079#issuecomment-1169377037

The alternative we have would be to set the typeVersions as suggested here:
https://github.com/microsoft/TypeScript/issues/33079#issuecomment-1169419172

I am open to whichever solution you guys favor. I am convinced we need the double export because we do not want to maintain two packages going forward with this plugin.

A follow up which would be nice would be to update the eslint rule to allow non-root internal imports if the import targets an entry point of the imported package. I will create the issue if this PR gets approved

### CODE REVIEW

**Review by adriencaccia — CHANGES_REQUESTED** [review]
Great work :rocket:

Your plan for the `eslint` import rule is fine with me

File: tsconfig.json [hunk 1]
```diff
@@ -5,7 +5,7 @@
     "types": ["node", "jest"],
     "lib": ["es2020"],
     "removeComments": true,
-    "moduleResolution": "node",
+    "moduleResolution": "node16",
```
Comment (fargito) [inline tsconfig.json:8]: Why is this needed?

File: backend/orchestrator/serverless.ts [hunk 2]
```diff
@@ -11,18 +13,20 @@ import {
...
   frameworkVersion,
   configValidationMode: 'error',
   plugins: [
     'serverless-esbuild',
+    '@swarmion/serverless-cdk-plugin',
     '@swarmion/serverless-plugin',
     'serverless-iam-roles-per-function',
     'serverless-analyze-bundle-plugin',
   ],
+  serverlessConstruct: OrchestratorDynamodb,
```
Comment (GuillaumeLagrange) [inline backend/orchestrator/serverless.ts:29]: C O N S T R U C T

**Comment by GuillaumeLagrange** [comment]
We are closing this MR because we found a more pragmatic solution by exporting anything we need in the plugin class. We are opening a new MR and adressing all the comments unrelated to the import strategy over there (#213 )

This MR stays as his, but closed, in order to keep a diff in case we want to dig into the exports keyword in a near future

File: packages/serverless-cdk-plugin/README.md [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+```ts
+const serverlessConfiguration: AWS & CdkPluginConfig = {
+  service: 'test-app',
+  configValidationMode: 'error',
+  plugins: ['../../src'],
+  provider: {
+    name: 'aws',
+  },
+  // Other properties ...
+  serverlessConstruct: MyConstruct,
```
[the remark this answers is not in the document]
Comment (Eikix) [reply packages/serverless-cdk-plugin/README.md:73]: For now yes

File: backend/orchestrator/package.json [hunk 4]
```diff
@@ -23,6 +23,7 @@
   },
   "dependencies": {
     "@swarmion/orchestrator-contracts": "^0.8.8",
+    "@swarmion/serverless-cdk-plugin": "0.8.8",
```
[the remark this answers is not in the document]
Comment (Eikix) [reply backend/orchestrator/package.json:26]: ok

File: packages/serverless-cdk-plugin/README.md [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+
+### Use-case examples
+
+<img width="500" height="500" src="./static/lambda-sqs-lambda.drawio.svg"/>
+
+#### Code example:
+
+Here's a short code snippet to demonstrate how to use the plugin. Code snippets are taken from the [integrationTests](./integrationTests/) folder.
+
+Typically you would create a construct in the desired folder. For instance, a DynamoDB construct.
```
[the remark this answers is not in the document]
Comment (Eikix) [reply packages/serverless-cdk-plugin/README.md:37]: Ok

File: packages/serverless-cdk-plugin/package.json [hunk 6]
```diff
@@ -18,9 +18,9 @@
   "files": [
     "dist"
   ],
-  "main": "dist/cjs/index.js",
-  "module": "dist/esm/index.js",
-  "types": "dist/types/index.d.ts",
+  "main": "./dist/cjs/index.js",
+  "module": "./dist/esm/index.js",
+  "types": "./dist/types/index.d.ts",
```
[the remark this answers is not in the document]
Comment (Eikix) [reply packages/serverless-cdk-plugin/package.json:23]: Ok

File: backend/orchestrator/serverless.ts [hunk 7]
```diff
@@ -41,7 +45,7 @@ const serverlessConfiguration: AWS & ServerlessContracts = {
   },
   resources: {
     Description: 'Monorepo deployments orchestrator',
-    ...cdkResources,
+    // ...cdkResources,
```
[the remark this answers is not in the document]
Comment (Eikix) [reply backend/orchestrator/serverless.ts:48]: Ok

File: packages/serverless-cdk-plugin/README.md [hunk 8]
```diff
@@ -0,0 +1,93 @@
...
+
+- Developers that were using Cloudformation to provision resources that Serverless Framework does not manage can now use the AWS CDK. It is a more efficient and DX-friendly option.
+
+- Developers that were using the AWS CDK won't need to "transpile" their CDK constructs' code into Cloudformation before the Serverless Framework deploy step. This saves them a lot of boilerplate fatigue.
+
+## :raised_hands: Caveats :raised_hands:
+
+- The Serverless CDK bridge plugin is only available for Serverless framework Typescript users, as it relies on the existence of a `serverless.ts` file at the root of any service to function.
+
+- The plugin does not yet allow for referencing lambda resources coming from Serverless framework inside CDK construct code. This feature will be available in a second release.
```
[the remark this answers is not in the document]
Comment (Eikix) [reply packages/serverless-cdk-plugin/README.md:25]: Actually this feature is available now.

File: packages/serverless-cdk-plugin/README.md [hunk 9]
```diff
@@ -0,0 +1,93 @@
...
+
+---
+
+### Use-case examples
+
+<img width="500" height="500" src="./static/lambda-sqs-lambda.drawio.svg"/>
```
[the remark this answers is not in the document]
Comment (Eikix) [reply packages/serverless-cdk-plugin/README.md:31]: Sure, how about lambda-and-sqs-integration

**Pull request closed without merging**